daml-script/dump/src/main/scala/com/daml/script/dump/TreeUtils.scala

```diff
@@ -15,6 +17,21 @@ import scalaz.syntax.foldable._
 object TreeUtils {
   final case class Selector(i: Int)
 
+  /** Sort the active contract set topologically,
+    *  such that a contract at a given position in the list
+    *  is only referenced by other contracts later in the list.
+    */
+  def topoSortAcs(acs: Map[String, CreatedEvent]): List[CreatedEvent] = {
+    val graph: Graphs.Graph[String] = acs.mapValues(createdReferencedCids)
+    Graphs
+      .topoSort(graph)
+      .left
+      .map(msg => new RuntimeException(s"Encountered cyclic contract dependencies: $msg"))
```

Comment:
maybe throw an IllegalArgumentException instead? probably also worth documenting the throwing behavior in the scaladoc comment.